app/src/main/java/com/epfl/beatlink/repository/profile/ProfileRepositoryFirestore.kt

```diff
@@ -15,32 +20,25 @@ import kotlinx.coroutines.tasks.await
  *
  * @param db The Firestore instance to use for database operations.
  * @param auth The FirebaseAuth instance to use for authentication operations.
- * @param storage The FirebaseStorage instance to use for file storage operations.
  */
 open class ProfileRepositoryFirestore(
     private val db: FirebaseFirestore,
     private val auth: FirebaseAuth,
-    private val storage: FirebaseStorage
 ) : ProfileRepository {
+  private val collection = "userProfiles"
 
   private val collectionPath = "userProfiles"
 
-  override fun getUserId(): String? {
-    val userId = auth.currentUser?.uid
-    Log.d("PROFILE_GET_UID", "Current user ID: $userId")
-    return userId
-  }
-
   override suspend fun fetchProfile(userId: String): ProfileData? {
     return try {
-      val snapshot = db.collection(collectionPath).document(userId).get().await()
-      val profilePictureUrl = snapshot.getString("profilePicture")
+      val snapshot = db.collection(collection).document(userId).get().await()
+      val profilePicture = snapshot.getString("profilePicture")
```

Comment:
Instead of creating a new variable, you can directly add this initialization to line 41